app/src/main/java/com/bnyro/recorder/ui/components/PlayerController.kt

```diff
@@ -0,0 +1,185 @@
+                var tempSliderPosition by remember { mutableStateOf<Float?>(null) }
+                Slider(
+                    modifier = Modifier.weight(1f),
+                    value = tempSliderPosition ?: positionAndDuration.first.toFloat(),
+                    onValueChange = { tempSliderPosition = it },
+                    valueRange = 0f.rangeTo(
+                        positionAndDuration.second?.toFloat() ?: Float.MAX_VALUE
+                    ),
+                    onValueChangeFinished = {
+                        tempSliderPosition?.let {
+                            exoPlayer.seekTo(it.toLong())
+                        }
+                        tempSliderPosition = null
+                    }
+                )
+                Text(
+                    positionAndDuration.second?.let { DateUtils.formatElapsedTime(it / 1000) }
+                        ?: ""
+                )
+            }
+            ElevatedCard(
+                colors = CardDefaults.elevatedCardColors(
+                    containerColor = MaterialTheme.colorScheme.tertiaryContainer,
+                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer
+                ),
+                shape = CircleShape
+            ) {
+                val playState by isPlayingState()
+                IconButton(
+                    onClick = {
+                        playPause()
+                    }
+                ) {
+                    when (playState) {
+                        PlayerState.Buffer -> {
+                            CircularProgressIndicator(modifier = Modifier.size(24.dp))
+                        }
+
+                        PlayerState.Play -> {
+                            Icon(
+                                Icons.Default.Pause,
+                                contentDescription = "Pause"
+                            )
+                        }
+
+                        PlayerState.Pause -> {
+                            Icon(
+                                Icons.Default.PlayArrow,
+                                contentDescription = "Play"
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Player.isPlayingState(): State<PlayerState> {
+    return produceState(
+        initialValue = if (isPlaying) {
+            PlayerState.Play
+        } else {
+            PlayerState.Pause
+        },
+        this
+    ) {
+        val listener = object : Player.Listener {
+            override fun onIsPlayingChanged(isPlaying: Boolean) {
+                playbackState
+                value = if (isPlaying) {
+                    PlayerState.Play
+                } else {
+                    PlayerState.Pause
+                }
+            }
+        }
+        addListener(listener)
+        if (!isActive) {
+            removeListener(listener)
+        }
```

Comment:
```suggestion
        if (isActive) addListener(listener)
```

Reply:
We don't need to add the listener if we'd immediately remove it anyways.

Reply:
Maybe I'm stupid

I think this is how it should be
``` kotlin
addListener(listener)
while(isActive){
// Do nothing
}

 removeListener(listener)
```

Reply:
Oh, now I get it :joy: